src/features/resources/StepComponent.tsx

```diff
@@ -0,0 +1,462 @@
+                }}
+              ></Box>
+              <Box
+                sx={{
+                  mt: '30px',
+                  display: { xs: 'none', sm: 'flex' },
+                  alignItems: 'center',
+                }}
+              >
+                <Breadcrumbs
+                  aria-label="breadcrumb"
+                  separator={
+                    <Image
+                      src={'/icons/right-arrow.svg'}
+                      alt={'icon'}
+                      width={9}
+                      height={9}
+                      style={{ transform: 'translateY(20%)' }}
+                    />
+                  }
+                >
+                  {breadcrumbs.map((title, index) => (
+                    <BreadcrumbsHeading key={index} variant="body2">
+                      {title}
+                    </BreadcrumbsHeading>
+                  ))}
+                </Breadcrumbs>
+              </Box>
+              <Box sx={{ my: { xs: '20px', sm: '30px' } }}>
+                <ResourcesSubHeading>
+                  {breadcrumbs.length > 0 && (
+                    <span style={{ fontWeight: 'bold', marginTop: '8px' }}>
+                      {breadcrumbs[breadcrumbs.length - 1]}
+                    </span>
+                  )}
+                </ResourcesSubHeading>
+                <ResourcesPara variant="body1" sx={{ textAlign: 'start' }}>
+                  {currentStep?.stepHeading}
+                </ResourcesPara>
+              </Box>
+              <Box>
+                {currentStep?.stepNumber === 5 ? (
+                  <ResourcesTableContainer>
+                    <StyledTable sx={{ minWidth: 768 }}>
+                      <TableHead>
+                        <TableRow>
+                          {Array.isArray(options) &&
+                            options.map((option, index) =>
+                              option?.headings?.map((heading, headingIndex) => (
+                                <TableCell
+                                  key={`${index}-${headingIndex}`}
+                                  sx={{
+                                    fontWeight: 'bold',
+                                    '&:nth-last-child(1)': {
+                                      textAlign: 'right',
+                                    },
+                                  }}
+                                >
+                                  {heading}
+                                </TableCell>
+                              ))
+                            )}
+                        </TableRow>
+                      </TableHead>
+
+                      <TableBody>
+                        {Array.isArray(options) ? (
+                          options.map((option) => (
+                            <>
+                              {option?.data?.map((data, dataIndex) =>
+                                option?.name === 'Revision Notes' ? (
+                                  <TableRow key={dataIndex}>
```

Comment:
Instead of this, the two types of tables should be different components, and then have them rendered as per the use-case. Though this isn't a big problem but it makes the code-base cleaner and easier to understand down the line

Reply:
This is general best practice, that the code for a specific type of component should be in it's specific component. Since the two types of tables are fundamentally different, so it'd be better if you break them down into their own components